### DISCUSSION TITLE
Collection `duplicates` with callable and its return type

### DISCUSSION THREAD

**Discussion by gms8994** [laravel/framework#53488] [Q&A]
https://laravel.com/docs/11.x/collections#method-duplicates the source for this suggests that it can take a callable, but that the [callable should return a boolean value](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Collections/Collection.php#L310), though that callable is immediately passed to [map](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Collections/Collection.php#L316). It feels wrong for that to return bool, since (at least my understanding) is that the callable would be used to select the items in the collection that make an individual collection record unique. Am I misunderstanding the use-case for a callable being passed to duplicates?

**Comment by henzeb** [comment]
It would make sense if it were to return anything to return duplicates based on two keys. Have you tried that? because in that case the docblock is wrong.

**Reply by gms8994** [reply]
Yep, that's what led me down the path - I assumed the docblock was incorrect once I read the code, but wanted to make sure there wasn't something I was unaware of wrt its expected use. I've opened #53499 to fix this.
